packages/workflow/src/interfaces.ts

```diff
@@ -1,39 +1,140 @@
+   * Indexed information attached to the Workflow Execution
    */
-  namespace: string;
+  searchAttributes?: Record<string, SearchAttributeValue[]>;
 
   /**
-   * Task queue this Workflow is scheduled in
+   * Non-indexed information attached to the Workflow Execution
    */
-  taskQueue: string;
+  memo?: Record<string, unknown>;
+
+  /**
+   * Parent Workflow info (present if this is a Child Workflow)
+   */
+  parent?: ParentWorkflowInfo;
+
+  /**
+   * Result from the previous Run (present if this is a Cron Workflow or was Continued As New).
+   *
+   * An array of values, since other SDKs may return multiple values from a Workflow.
+   */
+  lastResult?: unknown[];
+
+  /**
+   * Failure from the previous Run (present when this Run is a retry, or the last Run of a Cron Workflow failed)
+   */
+  lastFailure?: TemporalFailure;
 
   /**
-   * Whether a Workflow is replaying history or processing new events
+   * Task queue this Workflow is executing on
    */
-  isReplaying: boolean;
+  taskQueue: string;
+
+  /**
+   * Less commonly used information
+   */
+  more: {
+    /**
+     * Namespace this Workflow is executing in
+     */
+    namespace: string;
+
+    /**
+     * Run Id of the first Run in this Execution Chain
+     */
+    firstExecutionRunId: string;
+
+    /**
+     * The last Run Id in this Execution Chain
+     */
+    continuedFromExecutionRunId?: string;
+
+    // TODO expose from Core
+    /**
+     * Time at which the Workflow Run started
+     */
+    // startTime: Date;
+
+    /**
+     * Milliseconds after which the Workflow Execution is automatically terminated by Temporal Server. Set via {@link WorkflowOptions.workflowExecutionTimeout}.
+     */
+    executionTimeout?: number;
+
+    /**
+     * Time at which the Workflow Execution expires
+     */
+    executionExpirationTime?: Date;
+
+    /**
+     * Milliseconds after which the Workflow Run is automatically terminated by Temporal Server. Set via {@link WorkflowOptions.workflowRunTimeout}.
+     */
+    runTimeout?: number;
+
+    /**
+     * Maximum execution time of a Workflow Task in milliseconds. Set via {@link WorkflowOptions.workflowTaskTimeout}.
+     */
+    taskTimeout: number;
+
+    /**
+     * Retry Policy for this Execution. Set via {@link WorkflowOptions.retry}.
+     */
+    retryPolicy?: RetryPolicy;
+
+    /**
+     * Starts at 1 and increments for every retry if there is a `retryPolicy`
+     */
+    attempt: number;
+
+    /**
+     * Cron Schedule for this Execution. Set via {@link WorkflowOptions.cronSchedule}.
+     */
+    cronSchedule?: string;
+
+    /**
+     * Milliseconds between Cron Runs
+     */
+    cronScheduleToScheduleInterval?: number;
+  };
+
+  /**
+   * These fields are non-deterministic. Workflow code should not do different things based on these fields.
+   */
+  unsafe: {
+    /**
+     * Whether a Workflow is replaying history or processing new events
+     */
+    isReplaying: boolean;
```

Comment:
I don't think this should be in workflow info. As a rule of thumb, I don't think workflow info should _ever_ change after the workflow is started from a user POV. I should be able to store the workflow info in a local variable and trust that it's immutable and doesn't change for the life of the workflow run. This property holds true in other SDKs I believe.

Reply:
You're right, we should move this from info, I originally missed this.